tools/autogenForSwaggers/sdkinfo.json

```diff
@@ -0,0 +1,384 @@
+        "dotNet": {
+            "name": "Media",
+            "commit": "3586e2989d502434c4f607dd38d40e46aabede5c"
+        }
+    },
+    {
+        "name": "monitor",
+        "sources": [ "compositeMonitorClient.json" ],
+        "dotNet": {
+            "namespace": "Microsoft.Azure.Management.Monitor",
+            "output": "Management.Monitor/Generated/Monitor"
+        }
+    },
+    {
+        "name": "arm-monitor",
+        "sources": [ "compositeMonitorManagementClient.json" ],
+        "dotNet": {
+            "namespace": "Microsoft.Azure.Management.Monitor.Management",
+            "output": "Management.Monitor/Generated/Management/Monitor"
+        }
+    },
+    {
+        "name": "arm-network",
+        "sources": [ "compositeNetworkClient.json" ]
+    },
+    {
+        "name": "arm-notificationhubs",
+        "sources": [ "2017-04-01/swagger/notificationhubs.json" ],
+        "dotNet": {
+            "name": "NotificationHubs"
+        }
+    },
+    {
+        "name": "arm-operationalinsights",
+        "sources": [ "compositeOperationalInsights.json" ],
+        "dotNet": {
+            "name": "OperationalInsights",
+            "test": "OperationalInsights.Test/OperationalInsights.Test.csproj",
+            "client": "OperationalInsightsManagementClient",
+            "ft": 1,
+            "autorest": "AutoRest.1.0.0-Nightly20170126"
+        }
+    },
+    {
+        "name": "arm-powerbiembedded",
+        "sources": [ "2016-01-29/swagger/powerbiembedded.json" ],
+        "dotNet": {
+            "name": "PowerBIEmbedded"
+        }
+    },
+    {
+        "name": "arm-recoveryservices",
+        "sources": [ "compositeRecoveryServicesClient.json" ],
+        "dotNet": {
+            "name": "RecoveryServices"
+        }
+    },
+    {
+        "name": "arm-recoveryservicesbackup",
+        "sources": [
+            "2016-08-10/swagger/operations.json",
+            "2016-12-01/swagger/backupManagement.json"
+        ],
+        "dotNet": {
+            "ft": 1,
+            "name": "RecoveryServices.Backup",
+            "client": "RecoveryServicesBackupClient"
+        }
+    },
+    {
+        "name": "arm-redis",
+        "sources": [ "2016-04-01/swagger/redis.json" ],
+        "dotNet": {
+            "name": "Redis",
+            "folder": "RedisCache",
+            "test": "RedisCache.Tests/RedisCache.Tests.csproj",
+            "autorest": "AutoRest.0.17.3"
+        }
+    },
+    {
+        "name": "arm-relay",
+        "sources": [ "2016-07-01/swagger/relay.json" ]
+    },
+    {
+        "name": "arm-resources",
+        "sources": [
+            "resources/2017-05-10/swagger/resources.json",
+            "locks/2016-09-01/swagger/locks.json",
+            "features/2015-12-01/swagger/features.json",
+            "subscriptions/2016-06-01/swagger/subscriptions.json",
+            "policy/2016-04-01/swagger/policy.json",
+            "links/2016-09-01/swagger/links.json"
+        ],
+        "isLegacy": true,
```

Comment:
The flag `isLegacy` is only apply for some specific sources where we need to delete `Genereted` folder once but run `AutoRest` several times for different inputs. I hope, we will get rid of such generations in the future, so I would use the **convention over configuration** principle for this particular flag. Anyway, `sdkinfo.lock.json` should contain all flags and parameters explicitly.